FHEM/14_SD_WS.pm

```diff
@@ -291,6 +293,42 @@ sub SD_WS_Parse($$)
 				hum        => sub {my (undef,$bitData) = @_; return (SD_WS_binaryToNumber($bitData,28,31) * 10) + (SD_WS_binaryToNumber($bitData,32,35));},
 				channel    => sub {my (undef,$bitData) = @_; return (SD_WS_binaryToNumber($bitData,38,39) );},
 			},
+		53 =>
+			{
+				# AURIOL AHFL 433 B2 IAN 314695 Message Format
+				# ----------------------------------------------------
+				# 0    4    8    12   16   20   24   28   32   36   40
+				# 0000 0111 0000 0000 1101 1111 0111 1010 0100 1110 00
+				# iiii iiii b?cc tttt tttt tttt hhhh hhh? ???? ssss ss
+				# i:  8 bit random id (changes on power-loss)
+				# b:  1 bit battery indicator (0=>OK, 1=>LOW)
+				# c:  2 bit channel, valid channels are 1-3
+				# t: 12 bit signed temperature, scaled by 10
+				# h:  7 bit humidity
+				# s:  6 bit checksum (sum over nibble 0 - 8)
+				# ?:  x bit unknown (bit 32-35 always 0100)
+				sensortype => 'Auriol IAN 314695',
+				model      => 'SD_WS_53_TH',
+				# prematch => sub {my $msg = shift; return 1 if ($msg =~ /^[0-9A-F]{11}$/); }, 							# prematch
+				prematch   => sub {my $msg = shift; return 1 if ($msg =~ /^[0-9A-F]{8}4[0-9A-F]{2}$/); },	# prematch 0700F276 4 A4
+				crcok      => sub	{	my (undef,$bitData) = @_;
+														my $sum = 0;
+														for (my $n = 0; $n < 36; $n += 4) {
+															$sum += SD_WS_binaryToNumber($bitData, $n, $n + 3)
+														}
+														if (($sum &= 0x3F) == SD_WS_binaryToNumber($bitData, 36, 41)) {
+															return 1;
+														} else {
+															Log3 $name, 3, "$name: SD_WS_53 Parse msg $msg - ERROR checksum $sum != " . SD_WS_binaryToNumber($bitData, 36, 41);
```

Comment:
Ach so meintest du das... $sum wird hier in der Logausgabe z.B. noch verwendet.

Reply:
okay hatte Tomaten auf den Augen :)